feat: course page and its components

pankaj-prog opened the pull request:
1. Course page banner
![Skill up - course page banner](https://user-images.githubusercontent.com/82706396/159170352-1deb1c95-420a-4a05-a0ef-099db35c817f.png)

2. Filter table + Course wrapper + Course card + floating button to scroll to to
![Skill up - course page main section](https://user-images.githubusercontent.com/82706396/159170357-f25ae366-8306-4b6e-8d68-51fda7a21386.png)

pankaj-prog committed:
feat: course page and its components

m26unkwn commented on src/pages/Courses/components/FilterTable/FilterTable.jsx:
```diff
@@ -0,0 +1,104 @@
+                  ) : (
+                    <input type="checkbox" id={type.id} />
+                  )}
+                  <label htmlFor={type.id}> {type.label} </label>
```
You can create a component for input and use it here.
The benefits of doing that is 
1. You don't have to write the ternary operator here.
2.  You can just directly passe the props everything thing will be handled by the component itself.

pankaj-prog replied:
Thank you for sharing this.

m26unkwn commented on src/pages/Courses/components/FilterTable/FilterTable.jsx:
```diff
@@ -0,0 +1,104 @@
+        label: "Technology",
+      },
+    ],
+  },
+  {
+    heading: "Ratings",
+    name: "rating",
+    inputType: "radio",
+    filterType: [
+      {
+        id: "four-and-above",
+        label: "4 stars and above",
+      },
+      {
+        id: "three-and-above",
+        label: "3 stars and above",
+      },
+      {
+        id: "two-and-above",
+        label: "2 stars and above",
+      },
+    ],
+  },
+  {
+    heading: "Sort By",
+    name: "sort",
+    inputType: "radio",
+    filterType: [
+      {
+        id: "high-to-low",
+        label: "Price - high to low ",
+      },
+      {
+        id: "low-to-high",
+        label: "Price - low to high",
+      },
+    ],
+  },
+];
+
```
You can put this filter data in separate file and export it  from there.

1. It will reduce the line of code in this particular file.

pankaj-prog replied:
Yeah i will move that to data folder

m26unkwn reviewed (approved):
Approach that  i consider which are good.
1, Creating the data  `{}`f  or the input data.


everything seems good to me and you can look over the comment that i have given.

saurabhchirde reviewed (approved):
The code looks good to me, didn't find any issue. 
Good to merge.

PuneetP16 commented on src/components/Navbar/Navbar.css:
```diff
@@ -1,5 +1,11 @@
 .header-wrapper {
   box-shadow: 0px 0px 12px -6px var(--BLACK);
+  position: sticky;
+  background-color: white;
+  z-index: 2;
+  top: 0;
```
Suggestion: Add z-index in variable so that it would make easy to manage all the z-indexes in one place.

example:
   ```
   :root {
    --z-index-1: 1;
    --z-index-2: 2;
    --z-index-3: 3;
    --z-index-4: 4;
    --z-index-5: 5;
    }
```

PuneetP16 commented on src/pages/Courses/Courses.jsx:
```diff
@@ -1,7 +1,18 @@
 import React from "react";
```
You can remove this line of code as it will work without it aswell.

PuneetP16 commented on src/pages/Courses/components/CoursesContainer/CoursesContainer.css:
```diff
@@ -0,0 +1,11 @@
+.courses-container {
+  display: grid;
+  grid-template-columns: repeat(auto-fill, minmax(16rem, 1fr));
+  position: sticky;
+  top: 64px;
```
Suggestion: I think position sticky is not required here. Code should work without it. Do check.

pankaj-prog replied:
Removed unused css

PuneetP16 commented on src/pages/Courses/components/FilterTable/FilterTable.jsx:
```diff
@@ -0,0 +1,104 @@
+        <button className="btn btn-link-primary">Clear</button>
+      </header>
+      <section className="price-group">
+        <h1 className="h5">Price</h1>
+        <form action="" className="price-form">
```
Suggestion: action attribute is empty in here, could remove it if its not required.

pankaj-prog replied:
removed

PuneetP16 reviewed (approved):
Great job on creating product listing page, UI looks decent. 

Mentioned few suggestions to work on and good to merge.

UI suggestion: A little gap could be provided between footer and main content 
![image](https://user-images.githubusercontent.com/91944618/159173604-9090c583-70d6-4308-aab5-054a6c637a2f.png)

pankaj-prog committed:
refactor: removed unused css.
feat: added input component for filter table
fix: add space between content and footer
